app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -47,38 +51,63 @@ class ManageObligationsController @Inject()(val checkSessionTimeout: SessionTime
                                             val retrieveBtaNavBar: NavBarPredicate,
                                             val obligationsView: ManageObligations,
                                             val auditingService: AuditingService,
+                                            val sessionService: SessionService,
                                             nextUpdatesService: NextUpdatesService)
                                            (implicit val ec: ExecutionContext,
                                             implicit override val mcc: MessagesControllerComponents,
                                             val appConfig: FrontendAppConfig) extends ClientConfirmedController
-  with FeatureSwitching {
+  with FeatureSwitching with IncomeSourcesUtils {
 
 
-  def showSelfEmployment(changeTo: String, taxYear: String, incomeSourceId: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+  def showSelfEmployment(changeTo: String, taxYear: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
     andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
     implicit user =>
-      handleRequest(
-        mode = SelfEmployment,
-        isAgent = false,
-        taxYear,
-        changeTo,
-        incomeSourceId
-      )
+      withIncomeSourcesFS {
+        sessionService.get(SessionKeys.incomeSourceId).flatMap {
+          case Right(incomeSourceIdMayBe) =>
+            incomeSourceIdMayBe match {
+              case Some(incomeSourceId) =>
+                handleRequest(
```

Comment:
same optimisation as per above comment, you can permit incomeSourceIdMaybe into handleRequest method + apply changes in the method as well